### DISCUSSION TITLE
Feature proposal: As a player, I want to quickly understand moved unit movement states at a glance, so that I don't have to recall all that information

### DISCUSSION THREAD

**Discussion by jonearley** [triplea-game/triplea#9025] [Feature Requests & Ideas]
## Motivation and context

During combat and non-combat stages, I have a hard time remembering what units moved or still have the potential for movement left. So, I started to wonder how I could take my TripleA turns more quickly and with less mental effort. I would rather spend my mental energy on strategy than looking up "hidden" unit information in the game to understand my options for strategy.

## Benefits and impact

To pull good ideas from another software discipline, in web development, there are Web Content Accessibility Guidelines that target lowing the need for users to remember information to make multi-step decisions (eg, [Understanding Success Criterion 3.3.8: Redundant entry](https://www.w3.org/WAI/WCAG22/Understanding/redundant-entry)). 

I think this proposed idea would deliver similar benefits:

- Users with cognitive disabilities experience short-term, working memory difficulty. Not having to repeatedly remember particular information reduces stress and the likelihood of mistakes.
- Users who experience difficulty forming new memories, recalling information, and other functions related to cognition can complete processes without having to unnecessarily rely on their memory.

tldr: If we could help reduce the need to recall needed decision-making game information from memory then more players will be able to participate and this will likely improve their experience and improve the accessibility of the game.

## Initial feature design idea

Add a setting, keyboard shortcut, or whatever is appropriate to show the unit movement information of moved units. We don't need to show the state if a unit hasn't moved because we know that information about the unit type (an infantry as 1 moved, a ship has either 2 or 3, ...) already.

### What could this look like?

Visually at the bottom right of the unit add a badge (maybe the one that shows now when you are _actively_ moving units) that displays the digit of the remaining movement (eg, 1/2, 0/2). Maybe the color could be inverted, such as black badge background with white text.

![triplea-unit-movement-feature-design-idea](https://user-images.githubusercontent.com/1678665/110988648-cf0ed980-833e-11eb-8567-f8e0d25ad38a.JPG)

Try and imagine that badge/bubble with the movement state fixed near the ship always after it moved. It could perhaps be styled a little differently from how it looks in this picture from the displays the movement live feature already in the game. Such as the inversed color idea above, or something to that effect.

### Tricky considerations or risks

Would this make the UI too cluttered? Maybe this could be optional for those that need/want it in settings.

How would this render next to the unit stack totals? Or units stacks that have mixed movements left? Not sure...

## Discussion

Does this feature proposal make sense? Are there any questions that you have that I could try and answer to clarify the idea?

I wonder since this proposal is pretty close to a feature already in the game that this idea is fairly feasible? Maybe?

**Comment by jonearley** [comment]
Hi, @DanVanAtta 😸 Your thoughts on this would be much appreciated.

**Comment by DanVanAtta** [comment]
Does the 'f' key to highlight current units or the 'unit scroller' that is in the bottom right change/impact this idea in any way?

**Reply by jonearley** [reply]
I'll admit I've yet to use this movement assistant, but I'll try it out with my present game to test and learn about it and I'll consider the change/impact then. Thanks for pointing those features out.

**Reply by DanVanAtta** [reply]
I've had thoughts to add an option to keep a permanent highlight on units that have movement left and then 'dim' them when they are moved. Getting the highlighting right is one of several challenges.

**Reply by jonearley** [reply]
Right, that seems to be the aspect that might be missing still. I would be happy to advise or visually design something if you could provide the conditions/states to consider?

I wonder if you have any experiments in the past you could share? Or maybe this idea is just in your memory.
